File: api/models/schemas.py
from pydantic import BaseModel
from typing import Literal, Optional, List
from uuid import UUID


class Chunk(BaseModel):
    content: str
    source: Optional[str] = None


class TrackRequest(BaseModel):
    question: str
    answer: str
    chunks: Optional[List[Chunk]] = []
    latency_ms: Optional[int] = None


class TrackResponse(BaseModel):
    id: UUID
    status: str = "ok"


class FeedbackRequest(BaseModel):
    evaluation_id: UUID
    rating: Literal[1, -1]  # 1=良い, -1=悪い
    comment: Optional[str] = None


class FeedbackResponse(BaseModel):
    id: UUID
    status: str = "ok"


class EvaluationSummary(BaseModel):
    id: UUID
    question: str
    answer: str
    chunks: Optional[List[Chunk]] = []
    latency_ms: Optional[int]
    auto_score_relevance: Optional[float]
    auto_score_faithfulness: Optional[float]
    auto_score_completeness: Optional[float]
    created_at: str


class StatsResponse(BaseModel):
    total: int
    avg_relevance: Optional[float]
    avg_faithfulness: Optional[float]
    avg_completeness: Optional[float]
    positive_feedback_rate: Optional[float]
    recent: List[EvaluationSummary]

File: api-worker/src/schemas.ts
import { z } from "zod";

export const ChunkSchema = z.object({
  content: z.string(),
  source: z.string().nullish(),
});
export type Chunk = z.infer<typeof ChunkSchema>;

export const TrackRequestSchema = z.object({
  question: z.string(),
  answer: z.string(),
  chunks: z.array(ChunkSchema).nullish(),
  latency_ms: z.number().int().nullish(),
});
export type TrackRequest = z.infer<typeof TrackRequestSchema>;

export const FeedbackRequestSchema = z.object({
  evaluation_id: z.string().uuid(),
  rating: z.union([z.literal(1), z.literal(-1)]),
  comment: z.string().nullish(),
});
export type FeedbackRequest = z.infer<typeof FeedbackRequestSchema>;

export const StatsQuerySchema = z.object({
  key_id: z.string().uuid(),
  offset: z.coerce.number().int().min(0).default(0),
  limit: z.coerce.number().int().min(1).max(100).default(20),
});
export type StatsQuery = z.infer<typeof StatsQuerySchema>;

export const CreateKeyRequestSchema = z.object({
  name: z.string().nullish(),
});
export type CreateKeyRequest = z.infer<typeof CreateKeyRequestSchema>;

export const SignupRequestSchema = z.object({
  email: z.string().email(),
  password: z.string().min(8).max(128),
});
export type SignupRequest = z.infer<typeof SignupRequestSchema>;

export const LoginRequestSchema = z.object({
  email: z.string().email(),
  password: z.string().min(1),
});
export type LoginRequest = z.infer<typeof LoginRequestSchema>;
